### ISSUE TITLE
Possible data races

### ISSUE THREAD

**Issue by ryancaicse** [mercury-hpc/mercury#563]
Hi, should thread-shared static variable `hg_progress_shutdown_flag ` be protected by locks?

https://github.com/mercury-hpc/mercury/blob/ce847b771def57e9d63f33313ed81149b366648b/Examples/src/example_rpc_engine.c#L50-L63

https://github.com/mercury-hpc/mercury/blob/ce847b771def57e9d63f33313ed81149b366648b/Examples/src/example_rpc_engine.c#L66-L84

**Comment by ryancaicse** [comment]
Could you take a look? Many thanks. @soumagne

**Comment by soumagne** [comment]
Yes you're right sorry, that's been low-priority for me on my list but that should be fixed to either use an atomic variable or use a lock.
